packages/google-pay-integration/src/google-pay-payment-strategy.ts

```diff
@@ -69,11 +69,12 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         await this._paymentIntegrationService.submitOrder();
 
         const nonce = await this._googlePayPaymentProcessor.getNonce(payment.methodId);
+        const deviceSessionId = await this._googlePayPaymentProcessor.getDeviceSessionId();
 
         try {
             await this._paymentIntegrationService.submitPayment({
                 ...payment,
-                paymentData: { nonce },
+                paymentData: { nonce, deviceSessionId },
```

Comment:
I am wondering if we have another payment data param in one of the other provider, then we should create almost the same as you did in this PR. On the other hand, we can think about creating generic method to avoid some problems in the future.

My idea is to create a method, that will be responsible to get all provider specific payment data in one method like: getPaymentData, that you can simply adjust in different providers.

```
const extraPaymentData = await this._googlePayPaymentProcessor.getPaymentData() || {};

...
paymentData: { nonce, ...extraPaymentData },
...
```

What do you think?